src/main/fc/boot.c

```diff
@@ -742,6 +742,22 @@ void configureScheduler(void)
 
     if (sensors(SENSOR_ACC)) {
         setTaskEnabled(TASK_ACCEL, true);
+        switch (gyroPeriodUs) {  // Switch statement kept in place to change acc rates in the future
```

Comment:
this terrible code is returning time and time again.

Either remove it or replace with series of `if`s with inequalities, so that all gyroPeriodUs get converted to reasonable value.
With this code there 500u is fast, 499us is slow ...